src/sync.h

```diff
@@ -138,14 +137,10 @@ class SCOPED_LOCKABLE UniqueLock : public Base
     void Enter(const char* pszName, const char* pszFile, int nLine)
     {
         EnterCritical(pszName, pszFile, nLine, Base::mutex());
-#ifdef DEBUG_LOCKCONTENTION
         if (!Base::try_lock()) {
```

Comment:
I am somewhat worried that the extra calls to `Base::try_lock` cause performance overhead in the non-lock contention case. We might want to gate *that too* on the log category being enabled.

Reply:
Thanks, wondering that as well (or alternatively, leave the directives in place and just improve the logging experience when `DEBUG_LOCKCONTENTION` is defined).

Reply:
Implemented your suggestion (thanks!)